db/db_iterator_test.cc

```diff
@@ -1696,6 +1696,64 @@ TEST_P(DBIteratorTest, IterSeekForPrevCrossingFiles) {
   }
 }
 
+TEST_P(DBIteratorTest, PrefixSameAsStartNotInDomain) {
+  Options options = CurrentOptions();
+  options.prefix_extractor.reset(NewFixedPrefixTransform(2));
+  options.disable_auto_compactions = true;
+  // Enable prefix bloom for SST files
+  BlockBasedTableOptions table_options;
+  table_options.filter_policy.reset(NewBloomFilterPolicy(10, true));
+  options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+  DestroyAndReopen(options);
+
+  ASSERT_OK(Put("a", "va1"));
+  ASSERT_OK(Put("aa1", "va1"));
+  ASSERT_OK(Put("bb1", "va1"));
+  ASSERT_OK(Put("bb2", "va2"));
+  ASSERT_OK(Put("bb3", "va3"));
+  ASSERT_OK(Put("e", "va3"));
+  ASSERT_OK(Flush());
+
+  {
+    ReadOptions ro;
+    ro.prefix_same_as_start = true;
+    std::unique_ptr<Iterator> iter(NewIterator(ro));
+
+    // Behavior of prefix_same_as_start = ture with seek key not in domain
```

Comment:
Yeah, this is one of the cases I was thinking about where we could consider making the iterator invalid.

Also, there's a typo ("true").